.github/workflows/facebook.yml

```diff
@@ -42,14 +42,18 @@ jobs:
       - name: Container Image Push
         run: docker push hubofallthings/$DATAPLUG:$GITHUB_SHA
 
+      - name: Extract Branch Name
+        run:  |
+          echo "BRANCH=$(echo ${GITHUB_REF})" >> $GITHUB_ENV
+
       - name: Container Image Push - Master
-        if: startsWith(github.ref, 'refs/heads/main')
+        if: contains(env.BRANCH, 'main')
```

Comment:
Why do you create an abstraction here? You could use directly `GITHUB_REF`

Reply:
since it is used in multiple places it means you only need to change it in a single spot
it also allows testing by printing the variable